plugins/woocommerce/includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -193,18 +193,34 @@ public function update( &$order ) {
 		// Update the order.
 		parent::update( $order );
 
-		// Fire a hook depending on the status - this should be considered a creation if it was previously draft status.
 		$new_status = $order->get_status( 'edit' );
 
-		if ( $new_status !== $previous_status && in_array( $previous_status, array( 'new', 'auto-draft', 'draft', 'checkout-draft' ), true ) ) {
+		// We need to remove the wc- prefix from the status for comparison and proper evaluation of new vs updated orders.
+		if ( strpos( $previous_status, 'wc-' ) === 0 ) {
+			$previous_status = substr( $previous_status, 3 );
+		}
+
+		if ( strpos( $new_status, 'wc-' ) === 0 ) {
+			$new_status = substr( $new_status, 3 );
+		}
+
+		$nonTriggeringOrderStatuses = array( 'new', 'auto-draft', 'draft', 'checkout-draft', 'refunded', 'failed', 'cancelled');
+
+		// This hook should be fired only if the new status is not one of the non-triggering statuses and the previous status was one of the non-triggering statuses.
+		if (
+			$new_status !== $previous_status
+			&& ! in_array( $new_status, $nonTriggeringOrderStatuses, true )
+			&& in_array( $previous_status, $nonTriggeringOrderStatuses, true )
```

Comment:
Does this work reliably if creating a new order from scratch? The `$previous_status` comes from the database, so if the order does not exist in the database yet will this break?

Reply:
We can implement a similar solution to what you suggested with HPOS. In this case, the `$previous_status` will be `false` if it fails to retrieve it from the database. By setting it to `new,` we can correctly trigger the hook if it lands outside the non-triggering statuses. Do you agree?

```php
// Also grab the current status so we can compare.
$previous_status = get_post_status( $order->get_id() );
// If the order doesn't exist in the DB, we will consider it as new.
if ( ! $previous_status ) {
	$previous_status = 'new';
}
```

I added this in https://github.com/woocommerce/woocommerce/pull/49098/commits/b733fa699ea423b3262b80d818519dc7d3dbd7b4 Let me know what you think